lib/errors.js

```diff
@@ -0,0 +1,41 @@
+'use strict'
+
+const util = require('util')
+const codes = {}
+
+/**
+ * ContentTypeParser
+*/
+createError('FST_ERR_CTP_ALREADY_PRESENT')
+createError('FST_ERR_CTP_INVALID_TYPE', TypeError)
+createError('FST_ERR_CTP_EMPTY_TYPE', TypeError)
+createError('FST_ERR_CTP_INVALID_HANDLER', TypeError)
+createError('FST_ERR_CTP_INVALID_PARSE_TYPE', TypeError)
+createError('FST_ERR_CTP_BODY_TOO_LARGE', RangeError)
+createError('FST_ERR_CTP_INVALID_MEDIA_TYPE')
+createError('FST_ERR_CTP_INVALID_CONTENT_LENGTH', RangeError)
+
+function createMessage (args = []) {
+  // only simple string or template literal
+  if (args.length === 1) return args[0]
+  return util.format.apply(null, args)
+}
+
+function createError (code, Base = Error) {
+  code = code.toUpperCase()
+  const errClass = class FastifyError extends Base {
+    constructor (...args) {
+      super(`Code: ${code}; ${createMessage(args)}`)
+    }
```

Comment:
is this adding another level in the stacktrace? can you please verify and and add a test for that?

Reply:
No, the constructor call is omitted. Do you know how can I test it reliable?

Reply:
@mcollina ping.